### PR TITLE
Components: Update JetpackHeader to support dark color scheme

### PR DESCRIPTION
**Pull request by tyxla**
When we introduced the color scheme update to Jetpack in #31141, we seem to have missed to update the partner logos. This PR takes care of that.

#### Changes proposed in this Pull Request

* Introduce a dark color scheme to the `<JetpackHeader />` component.
* Update all partner logos to support the dark color scheme.
* Add the dark color scheme specifically to the Jetpack login page (it's already always dark).
* Add the dark color scheme specifically to the Jetpack partner-branded connection pages (they're already always dark).
* Update the devdocs example to work with the dark color scheme (laser black).

#### Preview
Before:

After:

#### Testing instructions

* Checkout this branch and build Calypso.
* Go to http://calypso.localhost:3000/devdocs/design/jetpack-header and test with all color schemes.
* Verify logos look well.
* Consider the following partner IDs:
  * `51945` - Dreamhost
  * `49615` - Pressable
  * `57152` - MilesWeb
  * `58712` - Bluehost
  * `57836` - InMotion
  * `60178` - LiquidWeb
* For each partner ID, load `http://calypso.localhost:3000/log-in/jetpack?partner_id=12345` where `12345` is the partner ID, and verify it looks well.
* For each partner ID, load `http://calypso.localhost:3000/jetpack/connect?partner_id=12345` where `12345` is the partner ID, and verify it looks well.

### CODE REVIEW

**Comment by sirreal** [comment]
Much of this looks fine, but I'm concerned about modifying logos. Are the adapted logos provided or validated by the partners?

**Comment by tyxla** [comment]
With the exception of the LiquidWeb logo, I've used SVGs provided by @jeffgolenski. But that's a valid question and I'll let someone from @Automattic/jetpack-design chime in.

**Comment by jeffgolenski** [comment]
@sirreal I made sure I followed the branding guidelines for companies where I found branding guidelines. Some do not. I'll have to check with liquid web!

@tyxla thanks so much for wrangling this! I'd just check with @ebinnion to make sure there isn't any PR overlap.

Thank you all!

File: client/components/jetpack-header/style.scss [hunk 1]
```diff
@@ -6,3 +6,10 @@
 .jetpack-header > svg {
 	max-width: 100%;
 }
+
+.is-section-jetpack-connect,
+.is-jetpack-login {
+	.jetpack-header__partner-logo-plus .gridicon {
+		color: #bbc9d5;
+	}
+}
```
Comment (ebinnion) [inline client/components/jetpack-header/style.scss]: Nitpick: This file should end with an empty newline.

Comment (tyxla) [reply client/components/jetpack-header/style.scss]: Thanks, added an empty line in https://github.com/Automattic/wp-calypso/pull/31989/commits/cd645c3b9e54f1da7e96cb6a43e4f78ba6b5951d

**Comment by tyxla** [comment]
@jeffgolenski seems like Eric didn't get to it yesterday, so there's no overlap. Code is also approved, so I'll standby for your design ✅

**Comment by jeffgolenski** [comment]
@tyxla Just confirmed the liquid web color changes are fine. This is good to go from my perspective. 

Can either you or @ebinnion just upload a screenshot of one of the partner onboarding steps with these logos being used out in the wild. Just so I can share it.

**Comment by ebinnion** [comment]
> Can either you or @ebinnion just upload a screenshot of one of the partner onboarding steps with these logos being used out in the wild. Just so I can share it.

Yes sir 🖖 

Screen Shot 2019-04-03 at 3 31 09 PM
Screen Shot 2019-04-03 at 3 31 00 PM
Screen Shot 2019-04-03 at 3 30 46 PM

**Comment by jeffgolenski** [comment]
Beautiful. Thanks @ebinnion

**Comment by tyxla** [comment]
Thank you all! Merging.

**Pull request merged**